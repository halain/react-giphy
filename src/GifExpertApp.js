import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";


//funcional component... Componente en base de funciones 
export const GifExpertApp = () => {

    const [categories, setCategories] = useState(["Breaking Bad"]);

  // const handleAdd = ()=>{
  //     //setCategories([...categories, "HunterXHunter"]);
  //     setCategories( (cate) => [...cate, "HunterXHunter"] );
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      {/* se pasa la funcion setCategories al componente para que sea ejecutda desde el */}
        <AddCategory setCategories={setCategories}/>
      <hr />

      <ol>
        {categories.map((category) =>  (
            <GifGrid key= {category} category={category} />) 
        )}
      </ol>
    </>
  );

};
