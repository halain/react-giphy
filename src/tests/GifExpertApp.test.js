import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../GifExpertApp';


describe('pruebas en <GifExpertApp /> ', () => {
    

    test('debe mostarse correctamente ', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();
        
    });
    

    test('debe de mostrar una lista de categorias', () => {
       
        const categories = ['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GifExpertApp defautCategory= {categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    



});


