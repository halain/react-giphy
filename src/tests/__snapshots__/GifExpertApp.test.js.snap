// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp />  debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExpertApp />  debe mostarse correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Breaking Bad"
      key="Breaking Bad"
    />
  </ol>
</Fragment>
`;
